lib/teiserver/player/session.ex

```diff
@@ -1305,7 +1305,8 @@ defmodule Teiserver.Player.Session do
       # we get the same struct every time, possibly with nil values for some keys
       # but in the meantime, just defensively get the country
       country: Map.get(user, :country, "??"),
-      status: status
+      status: status,
+      roles: Teiserver.Player.TachyonHandler.convert_teiserver_roles_to_tachyon(user.roles)
```

Comment:
This should not be there. The Session should not have to bother about the protocol itself. The translation should happen in `Player.TachyonHandler`, this is where we go to and from internal representation to tachyon specifics.